Doc/glossary.rst

```diff
@@ -363,6 +386,28 @@ Glossary
       the :term:`cyclic garbage collector <garbage collection>` is to identify these groups and break the reference
       cycles so that the memory can be reclaimed.
 
+   data race
+      A situation where multiple threads access the same memory location
+      concurrently, at least one of the accesses is a write, and the threads
+      do not use any synchronization to control their access.  Data races
```

Comment:
Maybe:

```suggestion
      When user code creates a situation where multiple threads access the same memory location
      concurrently, and at least one of the accesses is a write.  Data races
```

Reply:
I like the text Lysandros has here. Maybe we can elaborate that data races in the CPython implementation are considered bugs but projects outside of CPython may have different approaches.